### DISCUSSION TITLE
Is there an easy way to create an index of all files inside a folder ?

### DISCUSSION THREAD

**Discussion by mathisgauthey** [squidfunk/mkdocs-material#6743] [Q&A] [answered]
Hey there ! I just switched from Hugo with the hugo-theme-stack theme to material mkdocs.

I'm still figuring a few things here. Basically I have this structure :

```
.
├── about/
│   └── index.md
├── assets/
│   ├── favicon.ico
│   └── logo.png
├── blog/
│   ├── posts
│   ├── .authors.yml
│   └── index.md
├── contact/
│   └── index.md
├── guides/
│   └── index.md
├── projects/
│   └── index.md
├── support/
│   └── index.md
├── tags/
│   └── index.md
├── index.md
└── .pages
```

That means I store my articles in : 
- Projects : That's my project portfolio
- Guides : That's where I share guides on the internet to help people
- Blog : That's where I share my thoughts and random stuff

I already love the Blog index page with the archive and categories.

I'd like to find a way to automate the index file of my `Projects` and `Guides` folder.

SOmething like that with images for projects : 

![image](https://github.com/squidfunk/mkdocs-material/assets/46576952/519471e4-b327-4fd7-97f4-fba745b9dbe7)

And maybe a simpler approach for guides : 

![image](https://github.com/squidfunk/mkdocs-material/assets/46576952/8f516be5-3612-4196-8b99-cb16146c76a6)

Is there an existing plugin to do such thing ?

Thanks in advance !

**Comment by alexvoss** [comment]
There are a number of ways of achieving this. You could use multiple instances of the blog plugin to get similar archive and categories pages for the guides and projects. Alternatively, with the [Insider edition](https://squidfunk.github.io/mkdocs-material/insiders/) you could use [tag listings](https://squidfunk.github.io/mkdocs-material/plugins/tags/#listing-configuration) to generate the lists of guides and projects. That might more sense for anything that does not have a temporal ordering like blog entries.

If you need more customization than is possible with these then you can write a [hook](https://www.mkdocs.org/user-guide/configuration/#hooks) to compile the necessary index pages yourself.

**Reply by mathisgauthey** [reply]
Hey there thanks for the input !

I think your first option might be the best suited for my usecase, reusing blog view.

One thing I didn't mention is that my categories are used to select between my three folder locations when importing from my obsidian notes. I could change it but I find it convenient and don't use other categories than `Projects`, `Guides` and `Blog`.

I'm hence deactivating the categories from the left side panel of the blog.

So one thing that would be awesome to have is something like the blog view but instead of having archive and categories, I'd have archive and tags for example 🤔 I don't know if it's doable.

And then I'll just have to find a way to add an image showcase for the project section by modifying the blog view somehow.

**Reply by alexvoss** [reply, marked as answer]
> So one thing that would be awesome to have is something like the blog view but instead of having archive and categories, I'd have archive and tags for example 🤔 I don't know if it's doable.

Have a look to see if the default [tag index functionality](https://squidfunk.github.io/mkdocs-material/plugins/tags/#config.tags_file) does what you need. With Insiders you would gain more control over the presentation of tag indexes if needed.

I think the requirement to include images is one that might be common enough that you could suggest it as a feature? I have certainly had the thought a number of times that this might be a nice thing to have. I would define an image in the meta-data, using the first image referenced as a fallback perhaps. The latter would require parsing the Markdown, though, so would be a bit more involved and/or fragile. Apart from making the image available in the index, the rest should be doable with CSS to get different arrangements.

**Reply by mathisgauthey** [reply]
That seems great, I'll play around with these bad boys, thanks ! I'll add a FR when I'm done with that one then !

**Reply by nielsnl68** [reply]
I was looking for the same thing. Reading the above document i did not understand what i am suppose to do other then adding the plugin parameters. Is there an example to look at?

**Reply by mathisgauthey** [reply]
> I was looking for the same thing. Reading the above document i did not understand what i am suppose to do other then adding the plugin parameters. Is there an example to look at?

I'll come to my computer and do a small write-up in a few hours of how I did it 🫡

Edit : I'm using the `awesome-pages` plugin. It allows for a more granular nav control without the need to specify each and every page.

For example, here's mine at the root of the `docs` folder : 

```txt
nav:
- Home:
  - index.md
- projects
- guides
- blog
- tags
- about
- contact
- support
```

This way, my blog is directly at the root of my website which is the home page, there's even a small personalized text before the blog list.

And then projects, guides and blogs can be accessed from the sidebar or the top bar 👍

Oh and the blog config :

```
  - blog:
      blog_dir: .
      post_url_format: "{slug}"
      blog_toc: true
      categories_url_format: "{slug}"
```

Second edit: I should mention that my website is accessible here if you want to take a look : https://mathisgauthey.github.io/

**Comment by maxi40m** [comment]
> Hey there ! I just switched from Hugo with the hugo-theme-stack theme to material mkdocs.
> 
> 
> 
> I'm still figuring a few things here. Basically I have this structure :
> 
> 
> 
> ```
> 
> .
> 
> ├── about/
> 
> │   └── index.md
> 
> ├── assets/
> 
> │   ├── favicon.ico
> 
> │   └── logo.png
> 
> ├── blog/
> 
> │   ├── posts
> 
> │   ├── .authors.yml
> 
> │   └── index.md
> 
> ├── contact/
> 
> │   └── index.md
> 
> ├── guides/
> 
> │   └── index.md
> 
> ├── projects/
> 
> │   └── index.md
> 
> ├── support/
> 
> │   └── index.md
> 
> ├── tags/
> 
> │   └── index.md
> 
> ├── index.md
> 
> └── .pages
> 
> ```
> 
> 
> 
> That means I store my articles in : 
> 
> - Projects : That's my project portfolio
> 
> - Guides : That's where I share guides on the internet to help people
> 
> - Blog : That's where I share my thoughts and random stuff
> 
> 
> 
> I already love the Blog index page with the archive and categories.
> 
> 
> 
> I'd like to find a way to automate the index file of my `Projects` and `Guides` folder.
> 
> 
> 
> SOmething like that with images for projects : 
> 
> 
> 
> ![image](https://github.com/squidfunk/mkdocs-material/assets/46576952/519471e4-b327-4fd7-97f4-fba745b9dbe7)
> 
> 
> 
> And maybe a simpler approach for guides : 
> 
> 
> 
> ![image](https://github.com/squidfunk/mkdocs-material/assets/46576952/8f516be5-3612-4196-8b99-cb16146c76a6)
> 
> 
> 
> Is there an existing plugin to do such thing ?
> 
> 
> 
> Thanks in advance !

**Reply by kamilkrzyskow** [reply]
Hello @maxi40m,
your message only shows the quoted copy of the OP, why is there no new content aside from that?

**Comment by v-schipka** [comment]
I'm a bit late, but in case someone else finds this, I'm throwing in these plugins:
- https://github.com/Andre601/mkdocs-pagenav-generator
- https://pypi.org/project/mkdocs-gen-nav-plugin/
- https://github.com/oprypin/mkdocs-literate-nav
